src/player.js

```diff
@@ -1811,13 +1811,15 @@ export default class Player extends FakeEventTarget {
         }
       });
       if (this.config.playback.playAdsWithMSE) {
-        this._eventManager.listen(this, AdEventType.AD_LOADED, (event: FakeEvent) => {
+        this._eventManager.listen(this, AdEventType.AD_LOADED, event => {
           if (event.payload.ad.linear) {
-            this._detachMediaSource();
+            this._eventManager.listenOnce(this, AdEventType.AD_BREAK_START, () => {
+              this._detachMediaSource();
+            });
           }
         });
-        this._eventManager.listen(this, AdEventType.AD_BREAK_END, this._attachMediaSource);
-        this._eventManager.listen(this, AdEventType.AD_ERROR, this._attachMediaSource);
+        this._eventManager.listen(this, AdEventType.AD_BREAK_END, () => this._attachMediaSource());
```

Comment:
was this a mistake? the target is this so the context should remain the same in the handler

Reply:
both correct wanted to be more explicit.